packages/legend-query-builder/src/stores/QueryBuilderChangeDetectionState.ts

```diff
@@ -14,24 +14,85 @@
  * limitations under the License.
  */
 
-import { RawLambda } from '@finos/legend-graph';
-import { hashObject } from '@finos/legend-shared';
+import {
+  type GeneratorFn,
+  LogEvent,
+  promisify,
+  assertErrorThrown,
+  IllegalStateError,
+} from '@finos/legend-shared';
+import { action, flow, makeObservable, observable, isComputedProp } from 'mobx';
 import type { QueryBuilderState } from './QueryBuilderState.js';
+import { keepAlive, type IDisposer } from 'mobx-utils';
+import { QUERY_BUILDER_CHANGE_DETECTION_EVENT } from '../graphManager/QueryBuilderSupportedFunctions.js';
 
 export class QueryBuilderChangeDetectionState {
-  querybuildState: QueryBuilderState;
-  queryHashCode = hashObject(new RawLambda(undefined, undefined));
-  isEnabled = false;
+  querybuilderState: QueryBuilderState;
+  hashCode?: string | undefined;
+  /**
+   * Keep the list of disposers to deactivate `keepAlive` for computed value of element hash code.
+   * See {@link start} for more details
+   */
+  private stateHashCodeKeepAliveComputationDisposers: IDisposer[] = [];
```

Comment:
let's remove this unnecessary mechanics. I would like to have a button in the status bar of query builder to show JSON/text diff of before and after query though (similar to the local changes panel in `Studio`). So it might be a good idea to keep track of the initial query.